frontend/src/components/TrashPageLoggedinView.tsx

```diff
@@ -0,0 +1,121 @@
+import { useState, useEffect } from "react";
+import { Col, Row, Spinner, Modal, Button } from "react-bootstrap";
+import { DeletedNote as DeletedNoteModel } from "../models/deletedNote";
+import * as NotesApi from "../network/notes_api";
+import styles from "../styles/TrashPage.module.css";
+import DeletedNote from "./DeletedNote";
+import useTrashCountStore from "../store/trashCountStore";
+
+const TrashPageLoggedinView = () => {
+  const [deletedNotes, setDeletedNotes] = useState<DeletedNoteModel[]>([]);
+  const [deletedNotesLoading, setDeletedNotesLoading] = useState(true);
+  const [showNotesLoadingError, setShowNotesLoadingError] = useState(false);
+
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+  const [noteToDelete, setNoteToDelete] = useState<DeletedNoteModel | null>(
+    null
+  );
+
+  const { decrement } = useTrashCountStore();
+
+  useEffect(() => {
+    async function loadNotes() {
+      try {
+        setShowNotesLoadingError(false);
+        setDeletedNotesLoading(true);
+        const deletedNotes = await NotesApi.fetcDeletedNotes();
+        setDeletedNotes(deletedNotes);
+      } catch (error) {
+        console.error(error);
+        alert(error);
+        setShowNotesLoadingError(true);
+      } finally {
+        setDeletedNotesLoading(false);
+      }
+    }
+    loadNotes();
+  }, []);
+
+  async function permaDeleteNote() {
+    if (!noteToDelete) return;
+    try {
+      setDeletedNotes(
+        deletedNotes.filter(
+          (existingNote) => existingNote._id !== noteToDelete._id
+        )
+      );
+      decrement();
+      await NotesApi.permaDeleteNote(noteToDelete._id);
```

Comment:
Personal opinion: I'd refrain from connecting buttons to the functions that directly make requests to backend. I'd gather all the functions for fetch requests in one file and create a click handler to validate the data, call the function that fetches the endpoint and then return the data if validated properly.